tools/yaml-templates/ios-test.yml

```diff
@@ -0,0 +1,91 @@
+parameters:
+  - name: 'iOSAppHostingSdkGitPath'
+    default: none
+    type: string
+
+steps:
+  - checkout: self
+  # branch should be changed after test
+  - checkout: ${{ parameters.iOSAppHostingSdkGitPath }}@refs/heads/kx/pipeline_connection_preparation
+    path: iOSHost
+    persistCredentials: true
+
+  - task: InstallSSHKey@0
+    displayName: 'Install an SSH key for OneDSTelemetry'
```

Comment:
>'Install an SSH key for OneDSTelemetry' [](http://example.com/codeflow?start=17&length=39)

Can you say more about why this is needed?

Reply:
This is a ssh key required by iOS side because E2E Test would mock the full situations. For that, telemetry stuff would be used and this key is required. I do not know every detail on that but probably @baljesingh could help answering why we need that. Also, based on the discussion between me and Baljeet, I think he is working on the process to get rid of this SSH key or something like that.

Reply:
Yes, OneDSPackage is required for end-to-end testing of the telemetry system. We need this package at Orange side (testing app) not at HubSDK. 

https://github.com/microsoft/cpp_client_telemetry

To install this package, we use CocoaPods, which further requires access to the GitHub repository mentioned above. For accessing the repository, we need an SSH key. This is a temporary solution. We'll ship it with the package later to eliminate the need to download it every time.

Reply:
So this key is required to install the cocoapod containing the package?

Is it a concern that the key is just publicly visible in this repository now?

Reply:
@AE-MS we used the secure method to install the SSH key. 
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/install-ssh-key?view=azure-devops
This key is required so that cocoapod can access the MS Internal repo to install this package.

Reply:
@TrevorJoelHarris  As of now, we saved the public key in the pipeline as a variable(marked as secret). We followed the "Usage and best practices" section of the link mentioned above.
"_**We recommend that you don't pass in your public key as plain text to the task configuration. Instead, [set a secret variable](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?tabs=yaml%2cbatch&view=azure-devops&preserve-view=true#secret-variables) in your pipeline for the contents of your mykey.pub file**_"